### DISCUSSION TITLE
Is it true that AFL++ has portability to replace libFuzzer?

### DISCUSSION THREAD

**Discussion by junwha** [AFLplusplus/AFLplusplus#1630] [Q&A] [answered]
Hi, I'm going to fuzz the oss-fuzz targets using AFL++ in my local machine, and I want to use the LIB_FUZZING_ENGINE as `libAFLDriver.a`.
And as I know, AFL++ automatically detects the flag `-fsanitize=fuzzer` and replace it to `libAFLDriver.a`.

However, when I try to fuzz nginx and leveldb in OSS-fuzz, it shows me "undefined reference to '__afl__init'". 
I pre-checked it can build fuzzer with libfuzzer (-fsanitize=fuzzer), but if I set the CC and CXX as `afl-clang-fast` and `afl-clang-fast++`, it shows me an undefined reference errors. 
![image](https://user-images.githubusercontent.com/17183234/215469431-9ce0e6ce-ca1c-41d6-b679-f694ef8055a9.png)

Is there the steps I omitted? 

Thanks.

**Comment by vanhauser-thc** [comment, marked as answer]
you wrote down no steps that you did, so how would anyone know? :)

All you need to do is this:
`python infra/helper.py build_fuzzers OSS-FUZZ-TARGET --engine afl`

**Reply by junwha** [reply]
It was because I manually build the oss-fuzz image and omit some LLVM environment. Thank you for your help!:)
